### PR TITLE
Add model validations, associations, and tests

### PR DESCRIPTION
**Pull request by CPowell23**
- Add model_spec tests
- Add model validations and associations
- Add database migration to make category name unique at the database level

Additional test setup:
- Add 'rails' specification for update to simplecov
- bcrypt to convert password to password_digest
- database_cleaner gem and configuration
- configurations for shoulda matchers
- launchy for save_and_open page with capybara
- rack session access for stubbing logged in users

### CODE REVIEW

File: app/models/board.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Board < ApplicationRecord
+  validates :private, presence: true
+  validates :name, presence: true
+  validates :description, presence: true
+  validates :registered_user, presence: true
```
Comment (lucyconklin) [inline app/models/board.rb:5]: Do we need to validate description?

File: app/models/board.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  validates :category, presence: true
+
+  belongs_to :category
+  belongs_to :registered_user
+  has_many :pins
+  has_many :likes, as: :target
```
Comment (lucyconklin) [inline app/models/board.rb:11]: I'm really interested to see this target business in action.

File: app/models/pin.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Pin < ApplicationRecord
+  validates :description, presence: true
+  validates :image_url, presence: true
+  validates :article_url, presence: true
+  validates :board, presence: true
+
```
Comment (lucyconklin) [inline app/models/pin.rb:6]: So  if we set this up like Pinterest does, pins will no longer `belongs_to :board`

Instead they will `has_many :boards through :board_pins` ?

File: spec/models/pin_spec.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+describe Pin do
+  context "validations" do
+    it { is_expected.to validate_presence_of(:description) }
+    it { is_expected.to validate_presence_of(:image_url) }
+    it { is_expected.to validate_presence_of(:article_url) }
+    it { is_expected.to validate_presence_of(:board) }
+  end
+
+  context "associations" do
+    it { is_expected.to belong_to(:board) }
```
Comment (lucyconklin) [inline spec/models/pin_spec.rb:12]: and here

`it { is_expected.to have_many(:boards) }`

File: app/models/pin.rb (hunk 3 above)
Comment (CPowell23) [reply app/models/pin.rb:6]: As per our conversation earlier @wlffann @akintner , we decided to leave this as is.

File: app/models/board.rb (hunk 2 above)
Comment (CPowell23) [reply app/models/board.rb:11]: Can't wait to show it to you guys!

File: spec/models/pin_spec.rb (hunk 4 above)
Comment (CPowell23) [reply spec/models/pin_spec.rb:12]: See above

**Pull request merged**